docs/source/articles/testing-concepts.html.md

```diff
@@ -0,0 +1,116 @@
+
+
+## Tests & Decorators
+
+Workarea applications are Rails applications and therefore inherit the toolchain and patterns for [testing Rails apps](https://guides.rubyonrails.org/testing.html).
+Refer to that document for Rails testing fundamentals, such as Minitest, test cases, and assertions.
+
+However, a new Workarea application includes a broad feature set by default and therefore includes _tests_ for those features.
+You can run the platform's own tests (referred to here as _Workarea tests_, to differentiate them from application tests) from your application.
+( See [Run Tests](/articles/run-tests.html). )
+
+You can also extend the platform's test suite by writing _decorators_ for existing tests and writing your own tests (referred to here as _application tests_ to differentiate them from Workarea tests).
+( See [Decorate & Write Tests](/articles/decorate-and-write-tests.html). )
+
+When you extend Workarea with your own customizations and functionality, you should extend the test suite as well.
+As a general rule, you should __prefer writing new tests over decorating existing tests__.
+New tests are easier to reason about and run.
+( See [Run Tests](/articles/run-tests.html). )
+
+However, in some situtations decorating an existing test is necessary.
+__Decorate a Workarea test when your platform extensions have caused an existing test to fail__.
+In this case, decorate the test to make it pass again or to skip the test if your changes have caused it to become irrelevant.
+( See [Decorate & Write Tests](/articles/decorate-and-write-tests.html). )
+
+
+## Test Runners
+
+_Test runners_ are command line applications used to run tests (Workarea tests and application tests).
+
+Rails provides [its own test runners](https://guides.rubyonrails.org/testing.html#the-rails-test-runner), and Workarea adds test runners that make it more convenient to run Workarea tests.
+Knowing which test runner to use with which arguments can be confusing, so refer to the recipes in [Run Tests](/articles/run-tests.html).
+
+
+## Test Helper
+
+Every Rails test file begins by requiring the application _test helper_, a file located at `<application_root>/test/test_helper.rb`.
+This file boots the application into an appropriate test environment and bootstraps it for testing.
+This includes loading Rails' test help, `<railties>/lib/rails/test_help.rb`, and Workarea's test help, `<workarea-testing>/lib/workarea/test_help.rb`.
+
+You'll see this file required in the boilerplate for [writing your own tests](/articles/decorate-and-write-tests.html).
+
+
+## Test Case Types
+
+Workarea defines its own _test case types_, and all Workarea test cases inherit from one of those types.
+
+When [writing your own tests](/articles/decorate-and-write-tests.html), choose the appropriate test case type.
+Refer to the following table for brief descriptions, and click through to the sources to see the full implementation of each type.
+
+| Test Case Type | Description |
```

Comment:
I didn't go as far as originally planned this this (i.e. writing full descriptions in RDoc) because I ran out of time, but moving these into a table with source code links should make it easier to separate out this reference material in the future if you decide to do so.